src/chem/aeronet_aod2ioda.py

```diff
@@ -104,76 +97,92 @@ def add_data(infile):
     varDict = defaultdict(lambda: defaultdict(dict))
     outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-    loc_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-    var_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
-    units = {}
-    units['latitude'] = 'degree'
-    units['longitude'] = 'degree'
-    units['station_elevation'] = 'm'
-
-    # Define varDict variables
-    print('Define varDict variables')
-    for key, value in obsvars.items():
-        print(key, value)
-        varDict[key]['valKey'] = key, writer.OvalName()
-        varDict[key]['errKey'] = key, writer.OerrName()
-        varDict[key]['qcKey'] = key, writer.OqcName()
-
-    # Define loc_mdata
-    print('Define loc_mdata')
-    loc_mdata['latitude'] = np.array(f3['latitude'])
-    loc_mdata['longitude'] = np.array(f3['longitude'])
-    loc_mdata['station_elevation'] = np.array(f3['elevation'])
-    loc_mdata['surface_type'] = np.full((nlocs), 1)
-
-    c = np.empty([nlocs], dtype='S50')
-    c[:] = np.array(f3.siteid)
-    loc_mdata['station_id'] = writer.FillNcVector(c, 'string')
+    varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
 
-    # Define datetime
-    d = np.empty([nlocs], 'S20')
-    for i in range(nlocs):
-        d[i] = f3.time[i].strftime('%Y-%m-%dT%H:%M:%SZ')
-    loc_mdata['datetime'] = writer.FillNcVector(d, 'datetime')
+    obsvars = {'aerosol_optical_depth': ['aod_340nm', 'aod_380nm',
+                                         'aod_440nm', 'aod_675nm',
+                                         'aod_500nm', 'aod_870nm',
+                                         'aod_1020nm', 'aod_1640nm']}
+
+    AttrData = {
+        'converter': os.path.basename(__file__),
+    }
+
+    DimDict = {
+    }
+
+    VarDims = {
+        'aerosol_optical_depth': ['nlocs', 'nchans'],
+        'frequency': ['nchans'],
+        'sensor_channel': ['nchans']
+    }
 
-    # Define var_mdata
-    print('Define var_mdata')
-    var_mdata['frequency'] = writer.FillNcVector(frequency, 'float')
-    var_mdata['sensor_channel'] = writer.FillNcVector(aod_chan, 'integer')
+    for key, value in obsvars.items():
+        varDict[key]['valKey'] = key, iconv.OvalName()
+        varAttrs[key, iconv.OvalName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OvalName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OvalName()]['units'] = 'unitless'
+        varDict[key]['errKey'] = key, iconv.OerrName()
+        varAttrs[key, iconv.OerrName()]['_FillValue'] = -999.
+        varAttrs[key, iconv.OerrName()]['units'] = 'unitless'
+        varAttrs[key, iconv.OerrName()]['coordinates'] = 'longitude latitude station_elevation'
+        varDict[key]['qcKey'] = key, iconv.OqcName()
+        varAttrs[key, iconv.OqcName()]['_FillValue'] = -999
+        varAttrs[key, iconv.OqcName()]['coordinates'] = 'longitude latitude station_elevation'
+        varAttrs[key, iconv.OqcName()]['units'] = ''
```

Comment:
Please check the values being assigned to the units attributes. What we finally decided on is:

* SI units when appropriate as specified in the spreadsheet (eg, air_temperature -> 'K')
* For string variables, use an empty string (ie, '')
* For unit-less variables
    * Categorical variables (eg, PreQC) use "unitless"
    * Mathematical variables (eg, mixing ratio)
        * If hidden units (eg, mixing ratio) express them - eg, "kg kg-1"
        * If pure numeric use "1"

The reason I'm asking is that it seems to me that AOD is a pure numeric quantity (unitless ratio with physical meaning) and would therefore have units set to '1'.

Thanks!

Reply:
Will modify accordingly later. Thanks.
Bo

Reply:
Steve, I have modified the codes as suggested. Seems working fine that passed all 5 checks. Please let me know if further change is needed. Thanks. 
Bo